src/platform/w3dengine/client/w3ddisplay.cpp

```diff
@@ -933,9 +1673,594 @@ bool W3DDisplay::Is_LetterBox_Fading()
+        }
+
+        if (d) {
+            const Object *o = d->Get_Object();
+            Utf8String str4;
+            str4.Set("No-Name");
+
+            if (o) {
+                Utf8String name = o->Get_Name();
+
+                if (!name.Is_Empty()) {
+                    str4 = o->Get_Name();
+                }
+            }
+
+            str1.Format(L"Select Info: '%S'(%S) at (%.3f,%.3f,%.3f)",
+                d->Get_Template()->Get_Name().Str(),
+                str4.Str(),
+                d->Get_Position()->x,
+                d->Get_Position()->y,
+                d->Get_Position()->z);
+
+            const PhysicsBehavior *p = o->Get_Physics();
+
+            PhysicsTurningType turn;
+            if (p) {
+                turn = p->Get_Turning();
+            } else {
+                turn = TURN_NONE;
+            }
+
+            DrawableLocoInfo *info = d->Get_Loco_Info();
+            if (info) {
+                str2.Format(L"\nPhysics Info -- Turn: %d, Pitch(accel): %.3f(%.3f), Roll(accel): %.3f(%.3f)",
+                    turn,
+                    info->m_accelerationPitch,
+                    info->m_accelerationPitchRate,
+                    info->m_accelerationRoll,
+                    info->m_accelerationRollRate);
+                str1.Concat(str2);
+            }
+
+#ifdef GAME_DEBUG_STRUCTS
+            DebugDrawStats stats;
+
+            for (DrawModule **m = d->Get_Draw_Modules(); *m; m++) {
+                (*m)->Gather_Draw_Stats(&DebugDrawStats);
+            }
+
+            if (stats->Get_Draw_Calls() > 0) {
+                str2.Format(L"\ndraw calls: %d(+%d) sort meshes: %d skins: %d  bones: %d",
+                    stats->Get_Draw_Calls(),
+                    stats->Get_Extra_Draw_Calls(),
+                    stats->Get_Sort_Meshes(),
+                    stats->Get_Skins(),
+                    stats->Get_Bones());
+                str1.Concat(str2);
+            }
+#endif
+
+            const BitFlags<MODELCONDITION_COUNT> &flags = d->Get_Condition_State();
+            int newline = 0;
+
+            for (int i = 0; i < MODELCONDITION_COUNT; i++) {
+                if (flags.Test(i)) {
+                    str2.Format(L"%S ", BitFlags<MODELCONDITION_COUNT>::Get_Bit_Names_List()[i]);
+                    str1.Concat(str2);
+                    newline++;
+
+                    if (newline == 4) {
+                        newline = 0;
+                        str1.Concat(L"\n");
+                    }
+                }
+            }
+        }
+
+        m_displayStrings[14]->Set_Text(str1);
+    }
+#endif
+}
+
+void W3DDisplay::Draw_Debug_Stats()
+{
+    int size = 3;
+
+    for (int i = 0; i < 16; i++) {
+        m_displayStrings[i]->Draw(3, size, 0xFFFFFFFF, 0xFF000000);
+        int x;
+        int y;
+        m_displayStrings[i]->Get_Size(&x, &y);
+        size += y;
+    }
+}
+
+void W3DDisplay::Draw_Benchmark()
+{
+    m_benchmarkDisplayString->Draw(3, 20, 0xFFFFFFFF, 0xFF000000);
```

Comment:
m_benchmarkDisplayString is a array of 1, so for loop

Reply:
fixed.